src/IlluminaSampleSheet/V2/BclConvert/OverrideCycle.php

```diff
@@ -7,17 +7,36 @@ class OverrideCycle
     /** @var array<CycleTypeWithCount> */
     public array $cycles;
 
-    /** @param array<CycleTypeWithCount> $firstCycle */
-    public function __construct(array $firstCycle)
+    /** @param array<CycleTypeWithCount> $cycles */
+    public function __construct(array $cycles)
     {
-        $this->cycles = $firstCycle;
+        $this->cycles = $cycles;
     }
 
-    public function toString(): string
+    /** @param bool|null $isSecondIndexAndForwardDirection null if it is not the secondIndex, true if it is the secondIndex and forwardDirection, false if it is the secondIndex and reverseDirection */
+    public function toString(int $fillUpToMax, ?bool $isSecondIndexAndForwardDirection): string
     {
-        return implode('', array_map(
+        $countOfAllCycleTypes = $this->sumCountOfAllCycles();
+        assert($countOfAllCycleTypes <= $fillUpToMax, 'The sum of all cycle types must be less than or equal to the fill up to max value.');
```

Comment:
Seems like this should actually throw at runtime

Reply:
Currently, this isn't logically possible, so I'm adding a pure assertion for development purposes.